lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb

```diff
@@ -118,8 +118,8 @@ def self.create_service_provision_request(svc_template, options = nil)
     end
 
     def self.create_retire_request(obj)
-      result = obj.object_send(:make_retire_request, User.current_user)
-      MiqAeServiceModelBase.wrap_results(result)
+      obj_class = "#{obj.object_class.base_model.to_s.split('::').last}".constantize
```

Comment:
You will want to use `name` instead of `to_s`.  While they are likely the same right now only `name` is guaranteed to be consistent for this usage.  The `to_s` method could potentially return additional data that would break this logic.

Can you give an example of what classes requires the `.split('::').last` logic?  There are no tests that cover any of these.

Reply:
@tinaafitz dump